### PR TITLE
[4.0] Non-clickable module in a menu item

### PR DESCRIPTION
**Pull request by amitranjan2**
Pull Request for Issue #23540 

### Summary of Changes
Added `data-toggle` and `data-target`

### Testing Instructions

Navigate to following page from the admin page
Menus >> Main Menu >> Home

Open module assignment tab

try to click item

### Expected result
Modal open to edit

### Actual result

Modal not opening

### Documentation Changes Required
No

### CODE REVIEW

**Comment by amitranjan2** [comment]
@dgrammatiko Please review.

**Comment by infograf768** [comment]
Partially successful.
Although we now get the modal, Saving (or Save and Close) does not save the modifications of the module.

`TypeError: c.contents is not a function admin-item-edit_modules.min.js:1:1867`

debug on
`TypeError: iframe.contents is not a function admin-item-edit_modules.js:89:13`

**Comment by amitranjan2** [comment]
ok thanks, I will fix

**Comment by mvanvu** [comment]
/build/media_src/com_menus/js/admin-item-edit_modules.es6.js line 75
`iFrame.setAttribute('class', 'class="iframe jviewport-height70"');` is that right?

OR this should be `iFrame.setAttribute('class', 'iframe jviewport-height70');` ?

**Comment by dgrammatiko** [comment]
@amitranjan2 can you please use the event listener? There was a reason I did it like that in the first place: https://github.com/joomla/joomla-cms/blob/4.0-dev/build/media/legacy/js/bootstrap-init.js#L97

So your code essentially will never work...

**Comment by dgrammatiko** [comment]
> Although we now get the modal, Saving (or Save and Close) does not save the modifications of the module.

@infograf768 please test: 
https://github.com/joomla/joomla-cms/pull/23600

**Comment by amitranjan2** [comment]
`
>@amitranjan2 can you please use the event listener? There was a reason I did it like that in the first place: https://github.com/joomla/joomla-cms/blob/4.0-dev/build/media/legacy/js/bootstrap-init.js#L97

>So your code essentially will never work...`

Yes i tried with `modal.addEventListener('show.bs.modal', (ev) => {}` but its not executing when the modal is open.
Can you please give me example?
@dgrammatiko

**Comment by dgrammatiko** [comment]
@amitranjan2 here is some code, admin-item-edit_modules.es6.js:
```js
/**
 * @copyright  Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
 * @license    GNU General Public License version 2 or later; see LICENSE.txt
 */

Joomla = window.Joomla || {};

(() => {
  'use strict';

  const options = Joomla.getOptions('menus-edit-modules');

  if (options) {
    window.viewLevels = options.viewLevels;
    window.menuId = parseInt(options.itemId, 10);
  }

  document.addEventListener('DOMContentLoaded', () => {
    const baseLink = 'index.php?option=com_modules&amp;client_id=0&task=module.edit&tmpl=component&view=module&layout=modal&id=';
    const assigned1 = document.getElementById('jform_toggle_modules_assigned1');
    const assigned0 = document.getElementById('jform_toggle_modules_assigned0');
    const published1 = document.getElementById('jform_toggle_modules_published1');
    const published0 = document.getElementById('jform_toggle_modules_published0');
    const linkElements = [].slice.call(document.getElementsByClassName('module-edit-link'));
    const elements = [].slice.call(document.querySelectorAll('#moduleEditModal .modal-footer .btn'));

    if (assigned1) {
      assigned1.addEventListener('click', () => {
        const list = [].slice.call(document.querySelectorAll('tr.no'));

        list.forEach((item) => {
          item.style.display = 'table-row';
        });
      });
    }

    if (assigned0) {
      assigned0.addEventListener('click', () => {
        const list = [].slice.call(document.querySelectorAll('tr.no'));

        list.forEach((item) => {
          item.style.display = 'none';
        });
      });
    }

    if (published1) {
      published1.addEventListener('click', () => {
        const list = [].slice.call(document.querySelectorAll('.table tr.unpublished'));

        list.forEach((item) => {
          item.style.display = 'table-row';
        });
      });
    }

    if (published0) {
      published0.addEventListener('click', () => {
        const list = [].slice.call(document.querySelectorAll('.table tr.unpublished'));

        list.forEach((item) => {
          item.style.display = 'none';
        });
      });
    }

    if (linkElements.length) {
      linkElements.forEach((linkElement) => {
        linkElement.addEventListener('click', (event) => {
          debugger;
          const link = baseLink + event.target.getAttribute('data-module-id');
          const modal = document.getElementById('moduleEditModal');
          const body = modal.querySelector('.modal-body');
          const iFrame = document.createElement('iframe');

          iFrame.src = link;
          iFrame.setAttribute('class', 'class="iframe jviewport-height70"');
          body.innerHTML = '';
          body.appendChild(iFrame);

          modal.open();
        });
      });
    }

    if (elements.length) {
      elements.forEach((element) => {
        element.addEventListener('click', (event) => {
          const target = event.target.getAttribute('data-target');

          if (target) {
            const iframe = document.querySelector('#moduleEditModal iframe');
            const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
            iframeDocument.querySelector(target).click();
          }
        });
      });
    }
  });
})();
```

Also admin-module-edit.es6.js line 18:
```js
          const updPosition = document.getElementById('jform_position').value;
```

Also please **remove** all the changes in the markup:
```html
					data-toggle="modal"
						data-target="#moduleEditModal"
```
This is coupling things to Bootstrap and we don't want that!!!

**Comment by amitranjan2** [comment]
thanks, I will update.
@dgrammatiko

File: build/media_src/com_menus/js/admin-item-edit_modules.es6.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ Joomla = window.Joomla || {};
   }
 
   document.addEventListener('DOMContentLoaded', () => {
-    const baseLink = 'index.php?option=com_modules&amp;client_id=0&amp;task=module.edit&amp;tmpl=component&amp;view=module&amp;layout=modal&amp;id=';
+    const baseLink = 'index.php?option=com_modules&amp;client_id=0&task=module.edit&tmpl=component&view=module&layout=modal&id=';
```
Comment (Quy) [inline build/media_src/com_menus/js/admin-item-edit_modules.es6.js]:
```suggestion
    const baseLink = 'index.php?option=com_modules&client_id=0&task=module.edit&tmpl=component&view=module&layout=modal&id=';
```

**Comment by amitranjan2** [comment]
please test this @Quy

**Comment by amitranjan2** [comment]
@infograf768 Please review this
I think now it's working fine.
opening as well as saving both, working fine

**Comment by ghost** [comment]
Please resolve conflicting files so this PR can get tested at Worldwide Pizza, Bugs & Fun, October 19th

**Comment by ttplpoojak** [comment]
I have tested this item :white_check_mark: successfully on 64e2b30d3dd71375b04180e9e102894b13f0545c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23609">issues.joomla.org/tracker/joomla-cms/23609</a>.</sub>

**Comment by Nilesh5995** [comment]
I have tested this item :white_check_mark: successfully on 64e2b30d3dd71375b04180e9e102894b13f0545c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23609">issues.joomla.org/tracker/joomla-cms/23609</a>.</sub>

**Comment by alikon** [comment]
@amitranjan2 can you solve conflict please

**Comment by KishoriBKarale** [comment]
I have tested this item :white_check_mark: successfully on 64e2b30d3dd71375b04180e9e102894b13f0545c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23609">issues.joomla.org/tracker/joomla-cms/23609</a>.</sub>

**Comment by pdavila2709** [comment]
I can not apply the patch with the Joomla Patch Tester. This is the message:

Error: The file marked for modification does not exist: build/media_src/com_menus/js/admin-item-edit_modules.es6.js<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23609">issues.joomla.org/tracker/joomla-cms/23609</a>.</sub>

**Comment by infograf768** [comment]
@amitranjan2 please solve conflicts

**Comment by Quy** [comment]
Replaced with PR #27455<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23609">issues.joomla.org/tracker/joomla-cms/23609</a>.</sub>

**Pull request closed without merging**